dll/win32/shell32/shfldr.h

```diff
@@ -154,6 +157,11 @@ static __inline int SHELL32_GUIDToStringW (REFGUID guid, LPWSTR str)
 void SHELL_FS_ProcessDisplayFilename(LPWSTR szPath, DWORD dwFlags);
 BOOL SHELL_FS_HideExtension(LPCWSTR pwszPath);
 
+static inline BOOL IsIllegalFsFileName(PCWSTR Name)
+{
+    return StrIsNullOrEmpty(Name) || StrCSpnIW(Name, INVALID_FILETITLE_CHARACTERSW) < lstrlenW(Name);
```

Comment:
This would avoid calculating a strlen on the string:
```suggestion
    return StrIsNullOrEmpty(Name) || !!wcspbrk(Name, INVALID_FILETITLE_CHARACTERSW);
```
(or with `StrPBrkW`)